be/src/exec/pipeline/pipeline_driver_executor.cpp

```diff
@@ -310,24 +310,34 @@ void GlobalDriverExecutor::report_exec_state(QueryContext* query_ctx, FragmentCo
         query_exec_wall_time->set(query_ctx->lifetime());
     }
 
-    // Load channel profile will be merged on FE
-    auto* load_channel_profile = fragment_ctx->runtime_state()->load_channel_profile();
-    auto params = ExecStateReporter::create_report_exec_status_params(query_ctx, fragment_ctx, profile,
-                                                                      load_channel_profile, status, done);
-    auto fe_addr = fragment_ctx->fe_addr();
+    const auto& fe_addr = fragment_ctx->fe_addr();
     if (fe_addr.hostname.empty()) {
         // query executed by external connectors, like spark and flink connector,
         // does not need to report exec state to FE, so return if fe addr is empty.
         return;
     }
 
+    // Load channel profile will be merged on FE
+    auto* load_channel_profile = fragment_ctx->runtime_state()->load_channel_profile();
+    std::shared_ptr<TReportExecStatusParams> params;
+    {
+        // move profile memory to process, similar with SinkBuffer. the params will be released in ExecStateReporter
+        int64_t before_bytes = CurrentThread::current().get_consumed_bytes();
+        params = ExecStateReporter::create_report_exec_status_params(query_ctx, fragment_ctx, profile,
+                                                                     load_channel_profile, status, done);
+        int64_t delta = CurrentThread::current().get_consumed_bytes() - before_bytes;
+
+        CurrentThread::current().mem_release(delta);
+        GlobalEnv::GetInstance()->process_mem_tracker()->consume(delta);
+    }
+
```

Comment:
unrelated change ?

Reply:
The pipeline generates the memory requested by the profile and releases it in another thread. This causes the memory for query peak stats to keep growing